### PR TITLE
add validate to resource in can-i

### PR DESCRIPTION
**Pull request by xilabao**
```
#kubectl auth can-i get xxx
the server doesn't have a resource type "xxx"
```

### CODE REVIEW

**Comment by liggitt** [comment]
This could be a warning, but I'm not sure we should refuse to do the access check

**Comment by xilabao** [comment]
```
#kubectl auth can-i get xxx
Warning: the server doesn't have a resource type "xxx"
yes
```
the result is still `yes`?

**Comment by deads2k** [comment]
> This could be a warning, but I'm not sure we should refuse to do the access check

I'm ok with a warning.

**Comment by xilabao** [comment]
```
#./cluster/kubectl.sh auth can-i get xxx
Warning: the server doesn't have a resource type 'xxx'
yes

#./cluster/kubectl.sh auth can-i get pod.batch
Warning: the server doesn't have a resource type 'pod' in group 'batch'
yes
```
@liggitt @deads2k PTAL

File: pkg/kubectl/cmd/auth/cani.go [hunk 1]
```diff
@@ -188,6 +188,11 @@ func resourceFor(mapper meta.RESTMapper, resourceArg string) schema.GroupVersion
 		var err error
 		gvr, err = mapper.ResourceFor(groupResource.WithVersion(""))
 		if err != nil {
+			if len(groupResource.Group) == 0 {
+				fmt.Fprintf(o.Out, "Warning: the server doesn't have a resource type '%s'\n", groupResource.Resource)
```
Comment (deads2k) [inline pkg/kubectl/cmd/auth/cani.go]: make an `o.Err` please.

**Comment by deads2k** [comment]
nit, then lgtm.  tag me again when its updated

**Comment by xilabao** [comment]
updated. @deads2k

**Comment by deads2k** [comment]
@eparis how can @xilabao get added to the "test my PRs without asking" group?

**Comment by deads2k** [comment]
@k8s-bot ok to test

**Pull request merged**